src/rpdk/core/data_loaders.py

```diff
@@ -74,6 +74,33 @@ def make_resource_validator():
     return make_validator(schema)
 
 
+def make_resource_validator_with_additional_properties_check():
```

Comment:
We must also make sure that resource owners do not specify `additionalProperties` when `properties`/`patternProperties` are not specified.

Reply:
Will handle this in a different PR